CBLClient/Apps/CBLTestServer-Android/app/src/main/java/com/couchbase/CouchbaseLiteServ/server/Server.java

```diff
@@ -0,0 +1,210 @@
+import com.couchbase.CouchbaseLiteServ.server.RequestHandler.ExpressionRequestHandler;
+import com.couchbase.CouchbaseLiteServ.server.RequestHandler.FunctionRequestHandler;
+import com.couchbase.CouchbaseLiteServ.server.RequestHandler.QueryRequestHandler;
+import com.couchbase.CouchbaseLiteServ.server.RequestHandler.ReplicatorConfigurationRequestHandler;
+import com.couchbase.CouchbaseLiteServ.server.RequestHandler.ReplicatorRequestHandler;
+import com.couchbase.CouchbaseLiteServ.server.RequestHandler.ResultRequestHandler;
+import com.couchbase.CouchbaseLiteServ.server.RequestHandler.SelectResultRequestHandler;
+import com.couchbase.CouchbaseLiteServ.server.RequestHandler.SessionAuthenticatorRequestHandler;
+import com.couchbase.lite.Database;
+
+import com.couchbase.lite.LogDomain;
+import com.couchbase.lite.LogLevel;
+import com.google.gson.Gson;
+
+import org.nanohttpd.protocols.http.IHTTPSession;
+import org.nanohttpd.protocols.http.NanoHTTPD;
+import org.nanohttpd.protocols.http.response.IStatus;
+import org.nanohttpd.protocols.http.response.Response;
+import org.nanohttpd.protocols.http.response.Status;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class Server extends NanoHTTPD {
+
+    private static final String TAG = "com.example.hemant.coucbaselitesampleapp.serve";
+    public String uri;
+
+    public Method method;
+
+    public Map<String, String> header;
+
+    public Map<String, String> parms;
+
+    public Map<String, List<String>> parameters;
+
+    public Map<String, String> files;
+
+    public Map<String, List<String>> decodedParamters;
+
+    public Map<String, List<String>> decodedParamtersFromParameter;
+
+    public String queryParameterString;
+
+    public static final Memory memory = new Memory();
+
+    public Server(String ip, int port) throws IOException {
+        super(port);
+        Database.setLogLevel(LogDomain.ALL, LogLevel.DEBUG);
+        System.out.print("Running! Point your Mobile browser to http://" + ip + ":" + port + "/\n");
+    }
+
+    @Override
+    public Response handle(IHTTPSession session) {
+        String path = session.getUri();
+        String method = (path.startsWith("/") ? path.substring(1) : path);
+        // Get args from query string.
+        Map<String, String> rawArgs = new HashMap<>();
+
+        Args args = new Args();
+        try {
+            session.parseBody(rawArgs);
+        } catch (IOException e) {
+            e.printStackTrace();
+        } catch (ResponseException e) {
+            e.printStackTrace();
+        }
+        Map<String, Object> query = new Gson().fromJson(rawArgs.get("postData"), Map.class);
+        if (query !=null){
+            for (String key : query.keySet()){
+                String value = (String) query.get(key);
+                /*String value = null;
+                try {
```

Comment:
We can remove this code, if it's not needed?

Reply:
from which line number to which line number?

Reply:
I think @hrajput89  is talking about the commented code